lib/marked.js

```diff
@@ -77,7 +77,7 @@ block.normal = merge({}, block);
  */
 
 block.gfm = merge({}, block.normal, {
-  fences: /^ *(`{3,}|~{3,})[ \.]*(\S+)? *\n([\s\S]*?\s*)\1 *(?:\n+|$)/,
+  fences: /^ {0,3}(`{3,}|~{3,})([^`~\n]*)\n(?:|([\s\S]*?)\n)(?: {0,3}\1[~`]* *(?:\n+|$)|$)/,
```

Comment:
Seeing `[\s\S]*?` worries me:
* `\s` is the inverse of `\S`, so why isn't `[\s\S]` just `.`?
* `*` is already `0 or more`, so why is the `?` necessary?

Reply:
@hmhealey kindly educated me on the nuances of `.` not matching a newline, and `*?` changing the normally greedy semantics. My regular expression knowledge has definitely grown stale :)